backend/src/impl/analyses/significance_analysis.py

```diff
@@ -0,0 +1,120 @@
+import numpy as np
+from explainaboard.info import SysOutputInfo
+from explainaboard.metrics.metric import MetricStats
+from explainaboard.metrics.registry import get_metric_config_class
+from explainaboard.utils.typing_utils import unwrap
+
+
+def update_metric_config(system_output: SysOutputInfo) -> SysOutputInfo:
+    metric_configs = [
+        get_metric_config_class(metric_config_dict["cls_name"])(**metric_config_dict)
+        for metric_config_dict in system_output.metric_configs
+    ]
+    # update metric_configs in system_output
+    system_output.metric_configs = metric_configs
+
+    return system_output
+
+
+"""
+significance test based on bootstrapped resampling method, which are
+controlled by two major
+hyper-parameters:
+(1) n_samples: The number of bootstrapped samples
+(2) prop_samples: The ratio of samples to take every time
+
+"""
+
+
+def significance_test(
```

Comment:
Maybe we should consider implementing this within the ExplainaBoard SDK, as unpaired significance tests are also implemented there? https://github.com/neulab/ExplainaBoard/blob/main/explainaboard/metrics/metric.py#L302

That seems like it would be more consistent and we could also easily add pairwise tests to the SDK too.

Also, suggestion for the name: `pairwise_significance_test`

Reply:
Yes, as I have mentioned at the beginning of this PR,
```
I originally implemented this in ExplainaBoard SDK while one thing hindered me,

How to maintain metric_stats in ExplainaBoard SDK
Once we figure out this, we could easily introduce the significance test into ExplainaBoard SDK.
```

Reply:
Sorry, I missed this in the original PR!

Could you clarify what the issue is? I feel like it'd be as simple as moving the `significance_test()` function that you implemented here into the `metrics.py` file of ExplainaBoard, but I might be missing something.